File: training-data/erlang/binary_search.erl
-module(binary_search).
-export([search/2]).

search(Tuple, Target) ->
    go(Tuple, Target, 1, tuple_size(Tuple)).

go(_Tuple, _Target, Low, High) when Low > High -> not_found;
go(Tuple, Target, Low, High) ->
    Mid = (Low + High) div 2,
    Value = element(Mid, Tuple),
    if
        Value =:= Target -> {found, Mid};
        Value < Target -> go(Tuple, Target, Mid + 1, High);
        true -> go(Tuple, Target, Low, Mid - 1)
    end.

File: training-data/elixir/binary_search.exs
defmodule BinarySearch do
  def search(list, target) do
    tuple = List.to_tuple(list)
    do_search(tuple, target, 0, tuple_size(tuple) - 1)
  end

  defp do_search(_tuple, _target, low, high) when low > high, do: -1

  defp do_search(tuple, target, low, high) do
    mid = div(low + high, 2)
    value = elem(tuple, mid)

    cond do
      value == target -> mid
      value < target -> do_search(tuple, target, mid + 1, high)
      true -> do_search(tuple, target, low, mid - 1)
    end
  end
end

IO.inspect(BinarySearch.search([1, 3, 5, 7, 9, 11], 7))
IO.inspect(BinarySearch.search([1, 3, 5, 7, 9, 11], 4))
